src/ace/managers/memory.c

```diff
@@ -23,14 +25,20 @@ typedef struct _tMemEntry {
 
 static UWORD s_uwLastId = 0;
 static tMemEntry *s_pMemTail;
-static tFile *s_pMemLog;
 static ULONG s_ulChipUsage, s_ulChipPeakUsage, s_ulFastUsage, s_ulFastPeakUsage;
 
 //---------------------------------------------------------------- MEM ENTRY FNS
 
 static void _memEntryAdd(
 	void *pAddr, ULONG ulSize, UWORD uwLine, const char *szFile
 ) {
+	// Ozzyboshi discovered that memory allocation works without re-enabling
+	// the OS. After analyzing what's under the hood, it looks like there are only
+	// Forbid/Permit calls inside OS mem fns and the rest is just plain code.
+	// Still, OS mem fns can be patched by Snoopdos and/or newer OS versions,
+	// so I guess the safest approach is not to assume anything and wake OS up
+	// in case it needs other components on some exotic configs.
```

Comment:
I guess we don't care about the performance impact here, since the allocation itself will probably dominate?

Reply:
Allocations are slow by themselves, but there's a problem with user experience of OS wakeup.

The problem is that in order to allocate the memory you need to wake OS up currently, and this implies waiting for at least 2 frames in order to wake up an OS and the same to put it to sleep, and when OS is allive our interrupt handlers aren't working - specifically CIA timers needed for music playback. Now think of the loading between levels where you deallocate things. Skipping the OS wakeup would allow you to play back music in the meantime without glitches, now you either need to pause the playback or fade the volume out to mask the problem.

Still, I think the better way is to add support for interrupts when OS is alive, so things will be slow but working. Also better with the far future of making OS disabling completely optional.